src/main/java/org/cactoos/iterable/IterableOf.java

```diff
@@ -154,14 +158,22 @@ public boolean equals(final Object other) {
                     () -> other != null,
                     () -> Iterable.class.isAssignableFrom(other.getClass()),
                     () -> {
-                        final Iterable<?> compared = (Iterable<?>) other;
-                        final Iterator<?> iterator = compared.iterator();
-                        return new Unchecked<>(
-                            new And(
-                                (X input) -> input.equals(iterator.next()),
-                                this
-                            )
-                        ).value();
+                        final Iterable<X> compared = (Iterable<X>) other;
+                        boolean equals;
+                        try {
+                            equals = new Checked<>(
```

Comment:
@iiylll in that case let's use `ScalarWithFallback` instead of Checked so that we don't have to handle the exception catching.

Reply:
@victornoel Thanks. That's what I've been looking for